lib/adapters/class-vip-enterprise-search.php

```diff
@@ -20,141 +17,112 @@
-		if ( empty( $searched_facets ) ) {
-			return $query;
+	public function action__ep_valid_response( $response ): void {
+		if ( ! empty( $response['aggregations'] ) ) {
+			$this->parse_aggregations( $response['aggregations'] );
 		}
+	}
 
-		// Try to get the list of configured facets.
-		$configured_facets = $this->get_facet_config();
-		if ( empty( $configured_facets ) ) {
-			return $query;
+	/**
+	 * A callback for the ep_elasticpress_enabled filter hook. Overrides the
+	 * normal behavior for ElasticPress to determine if it is enabled to allow
+	 * for an empty search string, if allowable by the configuration on this
+	 * adapter.
+	 *
+	 * @param bool     $enabled  Whether ElasticPress is enabled for the query or not.
+	 * @param WP_Query $wp_query The WP_Query being examined.
+	 *
+	 * @return bool Whether ElasticPress should be active for the query or not.
+	 */
+	public function filter__ep_elasticpress_enabled( $enabled, $wp_query ) {
+		if ( $this->get_allow_empty_search()
+			&& $wp_query->is_search()
+			&& $wp_query->is_main_query()
+			&& ! is_admin()
+		) {
+			return true;
 		}
 
-		// Match searched facets against configured facets and augment the query accordingly.
-		foreach ( $searched_facets as $facet_label => $facet_terms ) {
-			// Skip any specified facets that are not configured.
-			if ( empty( $configured_facets[ $facet_label ]['type'] ) ) {
-				continue;
-			}
+		return $enabled;
+	}
 
-			// Loop over terms and add each based on type.
-			foreach ( $facet_terms as $facet_term ) { // phpcs:ignore WordPressVIPMinimum.Variables.VariableAnalysis.UnusedVariable
-				switch ( $configured_facets[ $facet_label ]['type'] ) {
-					case 'taxonomy':
-						$query['query']['function_score']['query']['bool']['must'][] = [
-							'terms' => [
-								'terms.' . $configured_facets[ $facet_label ]['taxonomy'] . '.slug' => [
-									$facet_term,
-								],
-							],
-						];
-						break;
-					case 'post_type':
-						$query['query']['function_score']['query']['bool']['must'][] = [
-							'terms' => [
-								'post_type.raw' => [
-									$facet_term,
-								],
-							],
-						];
-						break;
-				}
-			}
+	/**
+	 * A callback for the ep_indexable_post_types filter hook. Filters the list
+	 * of post types that should be indexed in ElasticPress based on what was
+	 * configured. If no restrictions were specified, uses the default list
+	 * (all public post types).
+	 *
+	 * @param array $post_types An associative array of post type slugs.
+	 *
+	 * @return array The modified list of post types to index.
+	 */
+	public function filter__ep_indexable_post_types( $post_types ) {
+
+		// Determine whether we should filter the list or not.
+		$restricted_post_types = $this->get_restricted_post_types();
+		if ( empty( $restricted_post_types ) ) {
+			return $post_types;
 		}
 
-		// TODO Write DSL for date faceting as configured.
+		// Rebuild the list of post types using the allowlist.
```

Comment:
I like the term "allow" better than "restrict" here.  Possibly update the function name to get_allowed_post_types and variable to $allowed_post_types for better clarity?

Reply:
I agree that this is somewhat confusing. The reason for the distinction is that if you don't specify a restriction, the plugins will automatically include all public post types, so by specifying a list, we are restricting it to just those post types that we are explicitly allowing.